mijia/src/bluetooth/introspect.rs

```diff
@@ -104,6 +107,12 @@ pub enum Access {
     Write,
 }
 
+pub async fn introspect(introspectable: &impl Introspectable) -> Result<Node, BluetoothError> {
```

Comment:
If you wanted to make this an extension trait, you just need concrete types (`Box<dyn Future>` rather than `impl Future`). https://github.com/dtolnay/async-trait is the best-practice way to do this.

I'm also happy to ship this as-is.

Reply:
Done.